### PR TITLE
need authentication support

### PR DESCRIPTION
**Pull request by tanaydin**
so i added.

### CODE REVIEW

**Comment by tanaydin** [comment]
need add authentication to kill process etc. too.

**Comment by hasegeli** [comment]
MongoClient is new in pymongo version 2.4 [1]. Is it possible to handle it without breaking pymongo 2.0 compatibility?

http://api.mongodb.org/python/current/api/pymongo/mongo_client.html

File: motop.py [hunk 1]
```diff
@@ -291,14 +293,24 @@ class Server:
...
-    def __init__ (self, name, address, hideReplicationOperations = False):
+    def __init__ (self, name, address, m_login, m_password, hideReplicationOperations = False):
         self.__name = name
         if ':' not in address:
             self.__address = address + ':' + str (self.defaultPort)
         else:
             self.__address = address
         self.__hideReplicationOperations = hideReplicationOperations
-        self.__connection = pymongo.Connection (address, read_preference = self.readPreference)
+        if pymongo.version_tuple > (2.3):
```
Comment (hasegeli) [inline motop.py:303]: This is comparing a tuple and a float, result will be true until Pymongo 3.*.

File: motop.py [hunk 2]
```diff
@@ -509,8 +529,14 @@ def booleanVariableTrueSections (self, variable):
...
         for section in self.sections ():
-            servers.append (Server (section, self.__configParser.get (section, 'address'),
-                                    self.__configParser.getboolean (section, 'hideReplicationOperations')))
+            address = self.__configParser.get (section, 'address')
+            try:
+                username = self.__configParser.get(section, 'username')
+                password = self.__configParser.get(section, 'password')
+            except:
+                username = None
+                password = None
```
Comment (hasegeli) [inline motop.py:538]: Try except is not good for expected configuration variables.

File: motop.py [hunk 3]
```diff
@@ -394,7 +406,15 @@ def explainQuery (self, databaseName, collectionName, **kwargs):
...
         """Kill operation using the "mongo" executable on the shell. That is because I could not make it with
         pymongo."""
-        os.system ('echo "db.killOp (' + str (opid) + ')" | mongo --host ' + self.__address)
+        if self.__mongo_username is not None and self.__mongo_password is not None:
+            e_username = "-u %s" % (self.__mongo_username)
+            e_password = "-p %s" % (self.__mongo_password)
+        else:
+            e_username = ""
+            e_password = ""
+        execute_string = "echo 'db.killOp (%s)' | mongo %s/admin %s %s" % (str(opid), self.__address, e_username, e_password)
```
Comment (hasegeli) [inline motop.py:415]: This is too long for one line. String formatting with % is depracated [1] at Python 3.

[1] http://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting

**Comment by tanaydin** [comment]
thanks for comments.

**Pull request merged**